### PR TITLE
Add create/delete_user() in quilt3.admin, docs, rm non-working stuff

### PR DESCRIPTION
**Pull request by sir-sigurd**
# Description

# TODO

- [ ] Unit tests
- [ ] Confirm that this change meets security best practices and does not violate the security model
- [x] Documentation
- [x] [Changelog](CHANGELOG.md) entry (skip if change is not significant to end users, e.g. docs only)

### CODE REVIEW

File: api/python/quilt3/admin.py [hunk 1]
```diff
@@ -1,117 +1,56 @@
...
+import typing as T
+
 from .session import get_registry_url, get_session
 
 
-def create_role(name, arn):
+def create_user(*, username: str, email: str):
     """
-    Create a new role in your registry. Admins only.
+    Create a new user in your registry.
```
Comment (akarve) [inline api/python/quilt3/admin.py:9]:
```suggestion
    Create a new user in the registry.
```

File: api/python/quilt3/admin.py [hunk 2]
```diff
@@ -1,117 +1,56 @@
...
     )
 
-    return response.json()
 
-
-def edit_role(role_id, new_name=None, new_arn=None):
+def delete_user(*, username: str):
     """
-    Edit an existing role in your registry. Admins only.
+    Delete user from your registry.
```
Comment (akarve) [inline api/python/quilt3/admin.py:27]:
```suggestion
    Delete user from registry.
```

File: api/python/quilt3/admin.py [hunk 3]
```diff
@@ -1,117 +1,56 @@
 """Provides admin-only functions for Quilt."""
```
Comment (akarve) [inline api/python/quilt3/admin.py:1]:
```suggestion
"""APIs for Quilt administrators. 'Registry' refers to Quilt stack backend services, including identity management."""
```

File: api/python/quilt3/admin.py [hunk 4]
```diff
@@ -1,117 +1,56 @@
...
-            role_id=role_id
-            )
-        )
-
 
-def get_role(role_id):
+def set_role(*, username: str, role_name: T.Optional[str]):
     """
-    Get info on a role based on its ID. Admins only.
+    Set which role is associated with a user.
```
Comment (akarve) [inline api/python/quilt3/admin.py:43]:
```suggestion
    Set the named Quilt role for a user.
```

File: api/python/quilt3/admin.py [hunk 5]
```diff
@@ -1,117 +1,56 @@
...
-        )
-
-    return response.json()
-
-
-def list_roles():
-    """
-    List configured roles. Admins only.
+        username (str): Username of user to update.
+        role_name (str): Role name to set for the user. Use `None` to unset role.
```
Comment (akarve) [inline api/python/quilt3/admin.py:47]: Does unassign mean "default role" or something else? Please clarify.
```suggestion
        role_name (str): Quilt role name assign to the user. Set a `None` value to unassign the role.
```

File: docs/api-reference/Admin.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+
+# quilt3.admin
+Provides admin-only functions for Quilt.
```
Comment (akarve) [inline docs/api-reference/Admin.md:3]:
```suggestion
Admin APIs for Quilt stacks.
```

File: docs/CHANGELOG.md [hunk 7]
```diff
@@ -18,6 +18,7 @@ Entries inside each section should be ordered by type:
 ## Python API
 
 ## CLI
+* [Added] `create_user()`, `delete_user()`, `set_role()` in `quilt3.admin` ([#3764](https://github.com/quiltdata/quilt/pull/3764))
```
Comment (sir-sigurd) [inline docs/CHANGELOG.md:21]: I don't mentioned removed functions because they were not documented and don't work for quite a long.

**Review by akarve — APPROVED** [review]

**Comment by drernie** [comment]
@sir-sigurd Is this close enough to try merging in for this release?

**Review by akarve — APPROVED** [review]

**Pull request merged**